btcwallet/btcwallet_test.go

```diff
@@ -0,0 +1,91 @@
+package btcwallet
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestBtcWallet_Deposit(t *testing.T) {
+	const (
+		startBalance        Bitcoin = 20
+		precision           int     = 3
+		depositErrorMessage string  = "the Bitcoins amount for the deposit must be positive"
+	)
+
+	testTable := []struct {
+		Source      Bitcoin
+		Expected    Bitcoin
+		ExpectedErr error
+	}{
+		{
+			Source:      23.456,
+			Expected:    43.456,
+			ExpectedErr: nil,
+		},
+		{
+			Source:      -0.234,
+			Expected:    startBalance,
+			ExpectedErr: errors.New(depositErrorMessage),
+		},
+	}
+
+	for _, tc := range testTable {
+		wallet := NewBtcWallet(startBalance)
+
+		if err := wallet.Deposit(tc.Source); err != nil {
+			if err.Error() != tc.ExpectedErr.Error() {
+				t.Error(err.Error())
+			}
```

Comment:
We can create a helper function to assert error/no error, also mark the helper with `t.Helper()`